src/Components/Server/src/Circuits/ServerAuthenticationStateProvider.cs

```diff
@@ -15,7 +15,7 @@ public class ServerAuthenticationStateProvider : AuthenticationStateProvider, IH
     /// <inheritdoc />
     public override Task<AuthenticationState> GetAuthenticationStateAsync()
         => _authenticationStateTask
-        ?? throw new InvalidOperationException($"{nameof(GetAuthenticationStateAsync)} was called before {nameof(SetAuthenticationState)}.");
+        ?? throw new InvalidOperationException($"{nameof(GetAuthenticationStateAsync)} was called before {nameof(SetAuthenticationState)}. You cannot call {nameof(GetAuthenticationStateAsync)} outside of the DI scope for a Razor component.");
```

Comment:
From context, I asume `SetAuthenticationState` is the start of "the DI scope for a Razor component"?

Reply:
> From context, I asume SetAuthenticationState is the start of "the DI scope for a Razor component"?

Not really, but the fact that you mention this underscores an aspect of the discussion we had about this yesterday. We debated whether people would naturally understand the terms "Blazor context" or "DI scope" (my guess is they will not). Unfortunately it is tough to give a precise definition without actually having to describe a bunch of the inner workings of DI and how it interacts with Blazor.

I think the second sentence of this error message is all that matters now, and we could avoid some confusion by removing the first sentence entirely. So it would simply state:

> You cannot call GetAuthenticationStateAsync outside of the DI scope for a Razor component.

Or if we wanted to be more helpful:

> Do not call GetAuthenticationStateAsync outside of the DI scope for a Razor component. Typically this means you can call it only within a Razor component or inside another DI service that is resolved for a Razor component.

Javier pointed out that there's at least one rare exception to this claim which is why I added the word *typically* here. It's probably not helpful to list the exceptions as that brings us back to having to explain inner workings of DI and its interaction with Blazor.

Reply:
I appreciate your input but I feel this addition is somewhat patronising.  What I am interested to understand is why technically this isn't doable?  Are Blazor and Razor doing completely different things with DI/scope, etc.?

Reply:
Hi @SteveSandersonMS 

That explanation is very helpful thank you.  I agree that dev's moving from .Net Framework (where webforms, MVC and API behaved in a much more similar fashion) will be caught out by this new approach.  I feel your first suggestion of the amendment would have massively helped us.  
It would be hugely helpful in future to have Blazor be able to behave like controllers in the sense they can't return raw data such as documents.  I believe Daniel Roth hinted this was incoming at the Build Conference?

Dave.

Reply:
```suggestion
        ?? throw new InvalidOperationException($"Do not call {nameof(GetAuthenticationStateAsync)} outside of the DI scope for a Razor component. Typically, this means you can call it only within a Razor component or inside another DI service that is resolved for a Razor component.");
```

I like the more helpful version. Can someone please approve this after I commit? Thanks!